src/kaocha/plugin.clj

```diff
@@ -16,22 +16,48 @@
     (catch java.io.FileNotFoundException e
       false)))
 
+(defn- try-require-info [n]
+  [(try-require n) n])
+
 (defn try-load-third-party-lib [plugin-name]
-  (if (qualified-keyword? plugin-name)
-    (when-not (try-require (symbol (str (namespace plugin-name) "." (name plugin-name))))
-      (try-require (symbol (namespace plugin-name))))
-    (try-require (symbol (name plugin-name)))))
+  (let [full-name (symbol (str (namespace plugin-name) "." (name plugin-name)))]
+    (if (qualified-keyword? plugin-name)
+      (if-let [full-result (try-require full-name)]
+        [[full-result full-name]]
+        [[false full-name] (try-require-info (symbol (namespace plugin-name))) ])
+      [(try-require-info (symbol (name plugin-name)))])))
 
 (defmulti -register "Add your plugin to the stack"
   (fn [name plugins] name))
 
 (defmethod -register :default [name plugins]
-  (output/error "Couldn't load plugin " name)
   (throw+ {:kaocha/early-exit 254} nil (str "Couldn't load plugin " name)))
 
 (defn register [plugin-name plugin-stack]
-  (try-load-third-party-lib plugin-name)
-  (-register plugin-name plugin-stack))
+  (let [ns-result (try-load-third-party-lib plugin-name)
+        successful-ns (map second (filter first ns-result))
+        failed-ns (map second (filter (complement first) ns-result))
```

Comment:
Similar comment as above, I'm having a hard time following this because it's not clear what `first` and `second` are. Can we write this in a clearer way?